pkg/cmd/experimental/syncgroups/schema1/memberextractor.go

```diff
@@ -0,0 +1,172 @@
+package schema1
+
+import (
+	"fmt"
+
+	"github.com/go-ldap/ldap"
+
+	ldaputil "github.com/openshift/origin/pkg/auth/authenticator/password/ldappassword/util"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/config"
+)
+
+// New builds a new GroupMemberExtractor using a schema-appropriate config
+func New(config LDAPSyncConfig) GroupMemberExtractor {
+	return GroupMemberExtractor{
+		ServerConfig:               config.CommonConfig.ServerConfig,
+		UserToIdentityMapping:      config.CommonConfig.LDAPUserToIdentityMapping,
+		SpecificGroupQueryConfig:   config.SpecificGroupQueryConfig,
+		SpecificUserQueryConfig:    config.SpecificUserQueryConfig,
+		GroupFieldAttributeMapping: config.GroupFieldAttributeMapping,
+		CachedUsers:                make(map[string]*ldap.Entry),
+		CachedGroups:               make(map[string]*ldap.Entry),
+	}
+}
+
+// GroupMemberExtractor extracts the member list of an LDAP group entry from an LDAP server
+// with first-class LDAP entries for groups
+type GroupMemberExtractor struct {
```

Comment:
This is satisfying several interfaces.  Tell me which ones in the godoc.  Also, to avoid bugs as you refactor (heavily), try somethign like `var _ interfaces.GroupExtractor = &GroupMemberExtractor{}`